py-polars/tests/unit/operations/test_cast.py

```diff
@@ -593,13 +593,39 @@ def test_strict_cast_string_and_binary(
         (pl.Datetime),
         (pl.Time),
         (pl.Duration),
-        (pl.Enum(["1"])),
+        (pl.String),
+        (pl.Categorical),
+        (pl.Enum(["1", "2"])),
```

Comment:
This PR technically doesn't cover this test but it's similar enough in scope; should have added these casts originally.

Reply:
If you're going to test the cast for every combination, then you can just add a separate mark for `dtype_in` and it will test all combinations.

Reply:
Didn't know you could do that, thanks for the tip!

Reply:
(updated with what I think you meant)

Reply:
Yep that's it!